docs/user/ppl/admin/catalog.rst

```diff
@@ -0,0 +1,88 @@
+.. highlight:: sh
+
+=================
+Catalog Settings
+=================
+
+.. rubric:: Table of contents
+
+.. contents::
+   :local:
+   :depth: 1
+
+Introduction
+============
+
+The concept of ``catalog`` is introduced to support the federation of SQL/PPL query engine to multiple data sources.
+This helps PPL users(plugins/customer) to leverage data from multiple data sources and derive correlation and insights.
+Catalog definition provides the information to connect to a datasource and also gives a name to them to refer in PPL commands.
+
+
+Definitions of catalog and connector
+====================================
+* Connector is a code component that adapts the query engine to a datasource. For example, Prometheus connector would adapt and help execute the queries to run on Prometheus data source.
+* Catalog is a construct to define how to connect to a datasource and which connector to adapt by query engine.
+
+Example Prometheus Catalog Definition ::
+
+    [{
+        "name" : "prometheus",
+        "connector": "prometheus",
+        "uri" : "http://localhost:9090",
+        "authentication" : {
+            "type" : "basicauth",
+            "username" : "admin",
+            "password" : "admin"
+        }
+    }]
+
+Configuring catalog in OpenSearch.
+====================================
+
+Catalogs are configured in opensearch keystore under `plugins.query.federation.catalog.config` key as they contain credential info.
+A json file containing array of catalog configurations should be injected into keystore with the above mentioned key.
+
+Command to add catalog.json file to OpenSearch Keystore ::
+
+    >>bin/opensearch-keystore add-file plugins.query.federation.catalog.config catalog.json
+
+Catalogs can be configured/overridden during either start up or while the opensearch is running.
+If we update catalog configuration during runtime, the following api should be triggered to update the query engine with the latest changes.
+
+Secure Settings refresh api::
+
+    >> curl --request POST \
+      --url http://{{opensearch-domain}}:9200/_nodes/reload_secure_settings \
+      --header 'content-type: application/json' \
+      --data '{"secure_settings_password":"{{keystore-password}}"}'
```

Comment:
what is the content been updated? what is secure_settings_password?

Reply:
If we want to reload secure settings in opensearch-keystore. We need to hit this reload api with key store password so that query engine can use the latest updated settings.